### PR TITLE
feature(classname-select) focused items are added to the canvas

### PR DESCRIPTION
**Pull request by enidemi**
**Problem:**
It's difficult to figure out what the select menu items are gonna do before selecting them.

**Fix:**
On menu item focus the hovered item is added to the classname list using transient canvas state.
I had to change how the selected value is created for the dropdown because of 2 issues:
- the fallback metadata props (that is used for controlled) are changing while in this transient state, triggering extra renders
- the classname attribute was also read from the jsx element found in instance metadata, which is now the transient jsx element, so I'm digging the projectContents to find the non-transient attributes. The attribute also had problems with memoization, now it's converted to a className string

**Commit Details:**
- actions to set/clear transient properties
- create transient canvas state
- changes about how the select menu values are made

### CODE REVIEW

File: editor/src/components/canvas/canvas-utils.ts [hunk 1]
```diff
@@ -2617,3 +2622,61 @@ export function getValidElementPathsFromElement(
...
+      editor,
+    )
+
+    let transientFilesState: TransientFilesState = {}
+    fastForEach(Object.values(editor.canvas.transientProperties) ?? [], (prop) => {
+      forUnderlyingTargetFromEditorState(
+        prop.elementPath,
+        updatedEditor,
+        (success, underlyingElement, underlyingTarget, underlyingFilePath) => {
+          transientFilesState[underlyingFilePath] = {
```
Comment (balazsbajorics) [inline editor/src/components/canvas/canvas-utils.ts:2668]: if I have two `editor.canvas.transientProperties` entries pointing to the same file, will they overwrite each other and the last one win out?

Comment (enidemi) [reply editor/src/components/canvas/canvas-utils.ts:2668]: nope, the transientproperties are applied to the jsx elements in a reduce in a few lines above this based on elementpath

Comment (enidemi) [reply editor/src/components/canvas/canvas-utils.ts:2668]: I should add some tests

Comment (balazsbajorics) [reply editor/src/components/canvas/canvas-utils.ts:2668]: or at least make yourself a TODO for tests :) but maybe not for today, as we are still figuring this feature out

File: editor/src/components/canvas/canvas-utils.ts [hunk 2]
```diff
@@ -1603,6 +1604,10 @@ export function produceCanvasTransientState(
     }
   }
 
+  if (editorState.canvas.transientProperties != null && editorState.canvas.dragState == null) {
+    transientState = createCanvasTransientStateFromProperties(editorState)
```
Comment (Rheeseyb) [inline editor/src/components/canvas/canvas-utils.ts:1608]: I think this should be moved to an `else if` clause inside the above block

File: editor/src/components/canvas/controls/classname-select.tsx [hunk 3]
```diff
@@ -413,22 +458,37 @@ export const ClassNameSelect: React.FunctionComponent = betterReactMemo('ClassNa
...
         }))
-  }, [classNameAttribute, classNameFromProps])
-
-  const isMenuEnabled = React.useMemo(
-    () =>
-      classNameAttribute == null ||
-      isJSXAttributeValue(classNameAttribute) ||
-      isJSXAttributeNotFound(classNameAttribute),
-    [classNameAttribute],
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (Rheeseyb) [inline editor/src/components/canvas/controls/classname-select.tsx:467]: This shouldn't be needed?

**Review by seanparsons — APPROVED** [review]

**Review by Rheeseyb — APPROVED** [review]

**Pull request merged**